Add an s to http if the host name contains 'api'

captKuk opened the pull request:
As the http port is closed on api-staging.materials.zone and the https port is working on api.materials.zone, an s is added to the http if the requested url contains "api".

oriyudilevich commented on mz_bokeh_package/utilities/environment.py:
```diff
@@ -37,15 +37,18 @@ def get_request_url(cls, endpoint: str) -> str:
+                add_an_s = "s"
 
-        return f"http://{host}/{endpoint}"
+        return f"http{add_an_s}://{host}/{endpoint}"
```
@captKuk the `add_an_s` variable is not necessary (and a bit "hacky"). I would just change this function to the following, and by this shifting the API traffic to the Kubernetes deployment:

```
env = cls.get_environment()
        add_an_s = ""

        if env == 'staging':
            host = 'https://api-staging.materials.zone/v1beta1'
        elif env == 'production':
            host = 'https://api.materials.zone/v1beta1'
        elif env == 'dev':
            host = os.getenv('API_HOST', 'https://api-staging.materials.zone/v1beta1')

        return f"http://{host}/{endpoint}"
```

Developers that are using the API_HOST environment variable will need to adapt their environment variable accordingly.

captKuk commented:
Hi @oriyudilevich, I've implemented your changes. 
Again: That means that the api* urls are completely equivalent to the *:5000 urls, right? There is no functionality lost when transitioning to the other url?

The tests are a bit silly, they could be written much nicer etc. But they hopefully fulfil the purpose. They also contain a function expected_status_code which is mostly there to run a weaker version of the tests, e.g. if they are run automatically on github and the credentials are not available. 
It essentially tests for staging and production:
 - that there are no redirects
 - that the user gets authenticated (if possible)
 - that a unknown user gets rejected.
If no credentials are provided (i.e. on github) the last test gets run twice.

oriyudilevich reviewed (changes requested):
@captKuk , I have the following comments about the tests:
- I don’t like that the tests are user dependent, that they depend on the keys you place in the environment variables. Tests should be always the same, reproducible. Also, we want to be able to run the tests in a server (e.g. in a deploy workflow). We have a few test users that we use for automated tests. My suggestion is to use these in the tests. So you can really hardcode the user_key and api_key in the code. As an example, the API has its own test user. It’s only staging I think. It’s hard-coded here: https://github.com/materialscloud/API/blob/staging/tests/test_users.py. I guess you can use this one or, better yet, create a separate test users for your tests here and hard-code their credentials in the repo so that it’s easy to find and maintainable (you'll see that I created a gmail account for test user so I don't associate it with a specific employee's email).
- Instead of duplicating the code for staging and production, you can pass the test an environment parameter, and then use parameters in the test to have the test loop over staging and environment. You do this using the @pytest.mark.parametrize decorator. We use it, for example, here: https://github.com/materialscloud/API/blob/staging/tests/tests_api/test_non_state_changing_endpoints_basic.py#L27
- Better not to have print inside the tests. If you want something to appear in the test report, you can read in the pytest documentation how to do this. There are several ways.
- Please add the pytest library to the setup.py file (under development, where the flake8 appears).

captKuk commented:
Pytest has been added to the setup.py file and the github action via rebasing the branch on the changes Misha made.

I also implemented a simple independent test in the directory tests/unit and removed the previous test.

MishaVeldhoen reviewed (changes requested):
I'm not 100% how it happened, but this branch now contains a number of changes that are already merged into master.

To remedy:
- undo all commits that are already on master (you may have to save your commits as patches, or so an interactive rebase)
- merge master into the cleaned up branch
- apply any changes you want to make on top of that

oriyudilevich commented on tests/unit/test_get_request_url.py:
```diff
@@ -0,0 +1,27 @@
+production_url = 'https://api.materials.zone/v1beta1' + '/'
+custom_api_host_url = 'http://staging.materials.zone:5000' + '/'
+
+# The order of the tests is important. set_env_to_development_with_own_host sets the environment variable
```
@captKuk I think this comment has to be updated, no? Cause I don't see a function called `set_env_to_development_with_own_host`. Better yet, I think it's nicer to remove the comment and simply clean the API_HOST environment variable in the beginning of the test using:
```
if 'API_HOST' in os.environ:
        del os.environ['API_HOST']
```
Then the order doesn't matter anymore.

oriyudilevich reviewed (approved):
@captKuk I'm approving the PR, but please take a look at my comment below and make the changes before merging to master.

After you merged the PR to master, you can tag the latest commit by following the following steps:
- make sure you are in the master branch (i.e. checkout master)
- run `git pull` to make sure you have the latest changes
- run `git tag v0.5.0' to add the tag locally
- run `git push origin --tags` to push the tag to Github (since just doing a push doesn't push tags).

You can read more about tags here: https://git-scm.com/book/en/v2/Git-Basics-Tagging.

After creating the tag, your changes are deployed and you can change to version v0.5.0 in the ingester.
